src/Features/Core/Portable/PullMemberUp/AbstractMembersPullerService.cs

```diff
@@ -299,6 +304,37 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
             }
 
             destinationEditor.ReplaceNode(destinationSyntaxNode, (syntaxNode, generator) => newDestination);
+
+            // add imports by combining source and destination imports, then taking out unneccessary
+            // imports that we don't need from the destination. If source and destination are the same document
+            // or have the same imports, this doesn't do anything
+            var destinationDocument = destinationEditor.GetChangedDocument();
+            var destinationRoot = await destinationDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var sourceImports = await GetImportsAsync(document, cancellationToken).ConfigureAwait(false);
+            var destinationImports = await GetImportsAsync(destinationDocument, cancellationToken).ConfigureAwait(false);
+            var destinationTrivia = GetFirstMemberTrivia(destinationRoot);
+            var addImportsService = destinationDocument.Project.LanguageServices.GetRequiredService<IAddImportsService>();
+            var semanticModel = await destinationDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            destinationDocument = destinationDocument.WithSyntaxRoot(addImportsService.AddImports(
+                semanticModel.Compilation,
+                destinationRoot,
+                null,
+                sourceImports,
+                destinationEditor.Generator,
+                options.PlaceSystemNamespaceFirst,
+                document.CanAddImportsInHiddenRegions(),
+                cancellationToken));
+
+            var removeImportsService = destinationDocument.Project.LanguageServices.GetRequiredService<IRemoveUnnecessaryImportsService>();
+            destinationDocument = await removeImportsService.RemoveUnnecessaryImportsAsync(
+                destinationDocument,
+                node => !IsValidUnnecessaryImport((TUsingOrAliasSyntax)node, destinationImports),
+                cancellationToken).ConfigureAwait(false);
+
+            destinationRoot = EnsureLeadingTriviaBeforeFirstMember(
```

Comment:
It seems like you are trying to maintain the trivia before the first member of the namespace/compilationUnit. 
could you give an example why you want to do this?

Reply:
In a case such as this:
```
// comment
class Foo
{
}
```
and we pull up a member from a file that has an unnecessary import:
```
using BazNs;
class Bar :  Foo
{
    public int field;
}
```
Adding the statement moves it on top of the using directive, and removing it as unnecessary removes the comment along with it. So, if we grab the leading trivia, and put it on afterwards, we retain the comment at the top of the class.

Reply:
: ) Could you put it into the comments?